src/battle_util.c

```diff
@@ -10310,6 +10310,49 @@ static uq4_12_t GetInverseTypeMultiplier(uq4_12_t multiplier)
     }
 }
 
+uq4_12_t GetTypeEffectiveness(struct Pokemon *mon, u8 moveType)
+{
+    uq4_12_t modifier = UQ_4_12(1.0);
+    u16 abilityDef = GetMonAbility(mon);
+    u16 speciesDef = GetMonData(mon, MON_DATA_SPECIES);
+    u8 type1 = gSpeciesInfo[speciesDef].types[0];
+    u8 type2 = gSpeciesInfo[speciesDef].types[1];
+
+    if (moveType != TYPE_MYSTERY)
+    {
+        MulByTypeEffectiveness(&modifier, MOVE_POUND, moveType, 0, type1, 0, FALSE);
+        if (type2 != type1)
+            MulByTypeEffectiveness(&modifier, MOVE_POUND, moveType, 0, type2, 0, FALSE);
+
+        if (abilityDef == ABILITY_LEVITATE && moveType == TYPE_GROUND)
+            modifier = UQ_4_12(0.0);
+        else if (abilityDef == ABILITY_WONDER_GUARD && modifier <= UQ_4_12(1.0))
+            modifier = UQ_4_12(0.0);
+        else if (abilityDef == ABILITY_FLASH_FIRE && moveType == TYPE_FIRE)
+            modifier = UQ_4_12(0.0);
+        else if (abilityDef == ABILITY_LIGHTNING_ROD && moveType == TYPE_ELECTRIC) // TODO: Add future Lightning Rod config
+            modifier = UQ_4_12(0.0);
+        else if (abilityDef == ABILITY_VOLT_ABSORB && moveType == TYPE_ELECTRIC)
+            modifier = UQ_4_12(0.0);
+        else if (abilityDef == ABILITY_MOTOR_DRIVE && moveType == TYPE_ELECTRIC)
+            modifier = UQ_4_12(0.0);
+        else if (abilityDef == ABILITY_MOTOR_DRIVE && moveType == TYPE_ELECTRIC)
+            modifier = UQ_4_12(0.0);
+        else if (abilityDef == ABILITY_WATER_ABSORB && moveType == TYPE_WATER)
+            modifier = UQ_4_12(0.0);
+        else if (abilityDef == ABILITY_DRY_SKIN && moveType == TYPE_WATER)
+            modifier = UQ_4_12(0.0);
+        else if (abilityDef == ABILITY_STORM_DRAIN && moveType == TYPE_WATER)
+            modifier = UQ_4_12(0.0);
+        else if (abilityDef == ABILITY_SAP_SIPPER && moveType == TYPE_GRASS)
+            modifier = UQ_4_12(0.0);
+        else if (abilityDef == ABILITY_EARTH_EATER && moveType == TYPE_GROUND)
+            modifier = UQ_4_12(0.0);
```

Comment:
Since they all do the same, I wonder if we should just collapse it into a single long condition with a bunch of ors

Reply:
Alright